web/src/main/java/se/inera/intyg/webcert/web/service/subscription/SubscriptionServiceImpl.java

```diff
@@ -129,25 +121,22 @@ public void acknowledgeSubscriptionWarning(WebCertUser webCertUser) {
         webCertUser.getSubscriptionInfo().getCareProvidersMissingSubscription().remove(selectedCareProvider.getId());
     }
 
+    @Override
+    public boolean isSubscriptionMissingWhenRequired(WebCertUser webCertUser) {
+        if (isFristaendeWebcertUser(webCertUser) && isSubscriptionRequired() && webCertUser.getSubscriptionInfo() != null) {
+            final var selectedCareProvider = webCertUser.getValdVardgivare();
+            final var missingSubscriptions = webCertUser.getSubscriptionInfo().getCareProvidersMissingSubscriptionUnmodifiable();
+            return selectedCareProvider != null && missingSubscriptions.contains(selectedCareProvider.getId());
+        }
+
+        return false;
+    }
+
     private void setSubscriptionActions(WebCertUser webCertUser, List<String> missingSubscriptions) {
         final var action = isSubscriptionRequired() ? SubscriptionAction.BLOCK : SubscriptionAction.WARN;
         webCertUser.getSubscriptionInfo().setSubscriptionAction(action);
         webCertUser.getSubscriptionInfo().setCareProvidersMissingSubscription(missingSubscriptions);
-    }
-
-    private void blockUsersWithoutAnySubscription(WebCertUser webCertUser, Collection<List<String>> allCareProviders,
-        List<String> careProvidersWithoutSubscription) {
-        if (isSubscriptionRequired()
-            && missingSubscriptionOnAllCareProviders(flatMapCollection(allCareProviders), careProvidersWithoutSubscription)) {
-            throw new MissingSubscriptionException(String.format("All care providers for user %s are missing subscription",
-                webCertUser.getHsaId()));
-        }
-    }
-
-    private boolean missingSubscriptionOnAllCareProviders(List<String> allCareProviders,
-        List<String> careProvidersWithoutSubscription) {
-        return allCareProviders.size() == careProvidersWithoutSubscription.size()
-            && allCareProviders.containsAll(careProvidersWithoutSubscription);
+        webCertUser.getSubscriptionInfo().setCareProvidersMissingSubscriptionUnmodifiable(List.copyOf(missingSubscriptions));
```

Comment:
Can we find a different solution to deal with the situation that a user already has been warned about missing subscription for a care provider.